### ISSUE TITLE
Plugin can't find other nodes: 'not enough master nodes'

### ISSUE THREAD

**Issue by akleiman** [elastic/elasticsearch-cloud-gce#53]
Hello,

I've been trying out the plugin for GCE, and I've had some trouble. ES 1.6, Plugin version 2.6.0, 3 nodes all running with `node.master=true` and `node.data=true`.

I've turned the plugin logging onto trace, and I see the following lines, repeated over and over:
```
[2015-07-03 00:24:00,406][TRACE][discovery.gce            ] [elasticsearch-cluster-node-1-node] starting to ping
[2015-07-03 00:24:03,409][TRACE][discovery.gce            ] [elasticsearch-cluster-node-1-node] full ping responses: {none}
[2015-07-03 00:24:03,409][DEBUG][discovery.gce            ] [elasticsearch-cluster-node-1-node] filtered ping responses: (filter_client[true], filter_data[false]) {none}
[2015-07-03 00:24:03,409][TRACE][discovery.gce            ] [elasticsearch-cluster-node-1-node] not enough master nodes [[[elasticsearch-cluster-node-1-node][0Qe5rL6JQieIx70b4tCxjA][elasticsearch-cluster-node-1.c.d1-dev.internal][inet[/10.240.55.30:9300]]{max_local_storage_nodes=1, master=true}]]
```

From the log messages, it doesn't even seem like the plugin code is being reached. In fact, if I put in garbage in the settings it makes no difference. Uninstalling the plugin raises an error, though.

Hopefully I haven't overlooked something in the documentation!

**Comment by dadoonet** [comment]
Did you set `minimum_master_nodes` to any value? What are your settings?

**Comment by akleiman** [comment]
Hi,

```
discovery.zen.minimum_master_nodes: 2
discovery.zen.ping.multicast.enabled: false
discovery.zen.fd.ping_interval: 1s
discovery.zen.fd.ping_timeout: 120s
discovery.zen.fd.ping_retries: 5

cloud.gce.project_id: d1-dev
cloud.gce.zone: us-central1-a
discovery.type: gce
```

**Comment by dadoonet** [comment]
How many nodes do you run?

**Comment by akleiman** [comment]
3 nodes, all with `node.master=true`.

**Comment by dadoonet** [comment]
The logs you pasted come from the first node, right? Can you paste as well your logs coming from the other nodes?

> From the log messages, it doesn't even seem like the plugin code is being reached.

Actually the plugin is started because `discovery.gce` package is logging things. So that looks correct.

So can you restart all the 3 nodes, and paste all 3 logs here?

Thanks!

**Comment by akleiman** [comment]
Sure:
node 1:
```
[2015-07-03 14:10:24,299][INFO ][node                     ] [elasticsearch-cluster-node-1-node] version[1.6.0], pid[4587], build[cdd3ac4/2015-06-09T13:36:34Z]
[2015-07-03 14:10:24,299][INFO ][node                     ] [elasticsearch-cluster-node-1-node] initializing ...
[2015-07-03 14:10:24,319][INFO ][plugins                  ] [elasticsearch-cluster-node-1-node] loaded [marvel, cloud-gce], sites [marvel]
[2015-07-03 14:10:24,368][INFO ][env                      ] [elasticsearch-cluster-node-1-node] using [1] data paths, mounts [[/var/lib/elasticsearch (/dev/sdb)]], net usable_space [186.6gb], net total_space [196.7gb], types [ext4]
[2015-07-03 14:10:26,164][DEBUG][discovery.gce            ] [elasticsearch-cluster-node-1-node] using ping.timeout [3s], join.timeout [1m], master_election.filter_client [true], master_election.filter_data [false]
[2015-07-03 14:10:26,842][INFO ][node                     ] [elasticsearch-cluster-node-1-node] initialized
[2015-07-03 14:10:26,843][INFO ][node                     ] [elasticsearch-cluster-node-1-node] starting ...
[2015-07-03 14:10:27,123][INFO ][transport                ] [elasticsearch-cluster-node-1-node] bound_address {inet[/0.0.0.0:9300]}, publish_address {inet[/10.240.16.218:9300]}
[2015-07-03 14:10:27,139][INFO ][discovery                ] [elasticsearch-cluster-node-1-node] dev.elasticsearch.cluster/vGdsxmeOSLiaMX6cZj90_g
[2015-07-03 14:10:27,141][TRACE][discovery.gce            ] [elasticsearch-cluster-node-1-node] starting to ping
[2015-07-03 14:10:30,150][TRACE][discovery.gce            ] [elasticsearch-cluster-node-1-node] full ping responses: {none}
[2015-07-03 14:10:30,151][DEBUG][discovery.gce            ] [elasticsearch-cluster-node-1-node] filtered ping responses: (filter_client[true], filter_data[false]) {none}
[2015-07-03 14:10:30,151][TRACE][discovery.gce            ] [elasticsearch-cluster-node-1-node] not enough master nodes [[[elasticsearch-cluster-node-1-node][vGdsxmeOSLiaMX6cZj90_g][elasticsearch-cluster-node-1.c.d1-dev.internal][inet[/10.240.16.218:9300]]{max_local_storage_nodes=1, master=true}]]
```

node 2:
```
[2015-07-03 14:10:24,120][INFO ][node                     ] [elasticsearch-cluster-node-2-node] version[1.6.0], pid[4592], build[cdd3ac4/2015-06-09T13:36:34Z]
[2015-07-03 14:10:24,120][INFO ][node                     ] [elasticsearch-cluster-node-2-node] initializing ...
[2015-07-03 14:10:24,136][INFO ][plugins                  ] [elasticsearch-cluster-node-2-node] loaded [marvel, cloud-gce], sites [marvel]
[2015-07-03 14:10:24,182][INFO ][env                      ] [elasticsearch-cluster-node-2-node] using [1] data paths, mounts [[/var/lib/elasticsearch (/dev/sdb)]], net usable_space [186.6gb], net total_space [196.7gb], types [ext4]
[2015-07-03 14:10:25,936][DEBUG][discovery.gce            ] [elasticsearch-cluster-node-2-node] using ping.timeout [3s], join.timeout [1m], master_election.filter_client [true], master_election.filter_data [false]
[2015-07-03 14:10:26,559][INFO ][node                     ] [elasticsearch-cluster-node-2-node] initialized
[2015-07-03 14:10:26,561][INFO ][node                     ] [elasticsearch-cluster-node-2-node] starting ...
[2015-07-03 14:10:26,934][INFO ][transport                ] [elasticsearch-cluster-node-2-node] bound_address {inet[/0.0.0.0:9300]}, publish_address {inet[/10.240.192.20:9300]}
[2015-07-03 14:10:26,959][INFO ][discovery                ] [elasticsearch-cluster-node-2-node] dev.elasticsearch.cluster/kFt6d2PjSCWoatD1VCj-jg
[2015-07-03 14:10:26,967][TRACE][discovery.gce            ] [elasticsearch-cluster-node-2-node] starting to ping
[2015-07-03 14:10:29,976][TRACE][discovery.gce            ] [elasticsearch-cluster-node-2-node] full ping responses: {none}
[2015-07-03 14:10:29,976][DEBUG][discovery.gce            ] [elasticsearch-cluster-node-2-node] filtered ping responses: (filter_client[true], filter_data[false]) {none}
[2015-07-03 14:10:29,976][TRACE][discovery.gce            ] [elasticsearch-cluster-node-2-node] not enough master nodes [[[elasticsearch-cluster-node-2-node][kFt6d2PjSCWoatD1VCj-jg][elasticsearch-cluster-node-2.c.d1-dev.internal][inet[/10.240.192.20:9300]]{max_local_storage_nodes=1, master=true}]]
```

node 3:
```
[2015-07-03 14:10:24,105][INFO ][node                     ] [elasticsearch-cluster-node-3-node] version[1.6.0], pid[4588], build[cdd3ac4/2015-06-09T13:36:34Z]
[2015-07-03 14:10:24,105][INFO ][node                     ] [elasticsearch-cluster-node-3-node] initializing ...
[2015-07-03 14:10:24,134][INFO ][plugins                  ] [elasticsearch-cluster-node-3-node] loaded [marvel, cloud-gce], sites [marvel]
[2015-07-03 14:10:24,191][INFO ][env                      ] [elasticsearch-cluster-node-3-node] using [1] data paths, mounts [[/var/lib/elasticsearch (/dev/sdb)]], net usable_space [186.6gb], net total_space [196.7gb], types [ext4]
[2015-07-03 14:10:26,058][DEBUG][discovery.gce            ] [elasticsearch-cluster-node-3-node] using ping.timeout [3s], join.timeout [1m], master_election.filter_client [true], master_election.filter_data [false]
[2015-07-03 14:10:26,805][INFO ][node                     ] [elasticsearch-cluster-node-3-node] initialized
[2015-07-03 14:10:26,805][INFO ][node                     ] [elasticsearch-cluster-node-3-node] starting ...
[2015-07-03 14:10:27,067][INFO ][transport                ] [elasticsearch-cluster-node-3-node] bound_address {inet[/0.0.0.0:9300]}, publish_address {inet[/10.240.226.52:9300]}
[2015-07-03 14:10:27,080][INFO ][discovery                ] [elasticsearch-cluster-node-3-node] dev.elasticsearch.cluster/en-bAX-dS_S8d5tG_agCvA
[2015-07-03 14:10:27,083][TRACE][discovery.gce            ] [elasticsearch-cluster-node-3-node] starting to ping
[2015-07-03 14:10:30,092][TRACE][discovery.gce            ] [elasticsearch-cluster-node-3-node] full ping responses: {none}
[2015-07-03 14:10:30,093][DEBUG][discovery.gce            ] [elasticsearch-cluster-node-3-node] filtered ping responses: (filter_client[true], filter_data[false]) {none}
[2015-07-03 14:10:30,093][TRACE][discovery.gce            ] [elasticsearch-cluster-node-3-node] not enough master nodes [[[elasticsearch-cluster-node-3-node][en-bAX-dS_S8d5tG_agCvA][elasticsearch-cluster-node-3.c.d1-dev.internal][inet[/10.240.226.52:9300]]{max_local_storage_nodes=1, master=true}]]
```

In all three logs, the `starting to ping` ... `not enough master nodes` block repeats endlessly after this.

**Comment by yaraju** [comment]
I [rewrote my script](https://storage.googleapis.com/prefab-pheonix-365.appspot.com/setup_es_1_5_noauth.sh) to install ES 1.5 and plugin version 2.5.0, and that works great, btw.

**Comment by schonfeld** [comment]
I'm going to venture a wild guess that this is happening cos someone decided to replace an important piece of code with a big "TODO" comment... 

https://github.com/elastic/elasticsearch-cloud-gce/blob/master/src/main/java/org/elasticsearch/discovery/gce/GceDiscovery.java#L49

**Comment by danielschonfeld** [comment]
cc @dadoonet :+1:

**Comment by dadoonet** [comment]
Closing as duplicate of https://github.com/elastic/elasticsearch/issues/13459

Thanks for opening this and sorry for the delay it's taking for fixing this. We'll see in #13459 how we can fix it and if needed fix branches 1.6 and 1.7.
In the meantime, it sounds like 2.5.0 version is working fine with 1.7.1 and probably with 1.6 versions.
